src/corecel/sys/KernelParamCalculator.device.hh

```diff
@@ -214,5 +218,29 @@ auto KernelParamCalculator::operator()(size_type min_num_threads) const
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Maximize L1 cache size if the kernel doesn't use static shared mem.
+ */
+template<class F>
+void KernelParamCalculator::set_carveout(F* kernel_func_ptr) const
+{
+#ifdef CELER_DEVICE_SOURCE
+    CELER_DEVICE_PREFIX(FuncAttributes) attr;
+    CELER_DEVICE_CALL_PREFIX(FuncGetAttributes(
+        &attr, reinterpret_cast<void const*>(kernel_func_ptr)));
+    if (!attr.sharedSizeBytes)
+    {
+        CELER_DEVICE_CALL_PREFIX(FuncSetAttribute(
+            kernel_func_ptr,
+            CELER_DEVICE_PREFIX(FuncAttributePreferredSharedMemoryCarveout),
+            CELER_DEVICE_PREFIX(SharedmemCarveoutMaxL1)));
```

Comment:
Looks like this is CUDA-only: https://cloud.cees.ornl.gov/jenkins-ci/blue/rest/organizations/jenkins/pipelines/Celeritas/branches/PR-1008/runs/2/nodes/18/steps/274/log/?start=0